src/objprint/frame_analyzer.py

```diff
@@ -102,3 +102,23 @@ def _splitlines_no_ff(self, source: str) -> List[str]:  # pragma: no cover
         if next_line:
             lines.append(next_line)
         return lines
+
+    def return_object(self, frame: Optional[FrameType]) -> bool:
+        current_frame = frame
+        while current_frame:
+            filename = current_frame.f_code.co_filename
+            if filename in ["<stdin>", "<console>"]:
+                return False
+            current_frame = current_frame.f_back
+        if frame is None:
+            return True
```

Comment:
I actually thought about this to avoid the type hint - I don't think this is better. Logically, it works. However, using `frame` before checking whether it's `None` feels weird. Let's still check `None` at the beginning of the function, and use that later, even if that means we need to put a type hint on a variable.

The code below is okay now.

Reply:
Sure!